packages/oscal-react-library/src/components/OSCALControl.tsx

```diff
@@ -1,32 +1,58 @@
 import React, { useEffect } from "react";
 import { styled } from "@mui/material/styles";
-import Card from "@mui/material/Card";
+import Card, { CardProps } from "@mui/material/Card";
 import CardContent from "@mui/material/CardContent";
 import Typography from "@mui/material/Typography";
 import Grid from "@mui/material/Grid";
 import OSCALControlLabel from "./OSCALControlLabel";
 import OSCALControlPart from "./OSCALControlPart";
 import OSCALControlModification from "./OSCALControlModification";
-import { OSCALAnchorLinkHeader } from "./OSCALAnchorLinkHeader";
+import { AnchorLinkProps, OSCALAnchorLinkHeader } from "./OSCALAnchorLinkHeader";
 import isWithdrawn from "./oscal-utils/OSCALCatalogUtils";
 import { propWithName } from "./oscal-utils/OSCALPropUtils";
 import { appendToFragmentPrefix } from "./oscal-utils/OSCALLinkUtils";
+import {
+  Control,
+  ImplementedRequirementElement,
+  SetParameterValue,
+  Alteration,
+} from "@easydynamics/oscal-types";
+import { EditableFieldProps } from "./OSCALEditableTextField";
 
-const OSCALControlCard = styled(Card, {
-  // https://github.com/mui/material-ui/blob/c34935814b81870ca325099cdf41a1025a85d4b5/packages/mui-system/src/createStyled.js#L56
-  shouldForwardProp: (prop) =>
-    !["childLevel", "withdrawn", "ownerState", "theme", "sx", "as"].includes(prop),
-})`
-  margin-top: 1em;
-  margin-bottom: 1em;
-  margin-left: ${(props) => (props.childLevel > 0 ? "1.5em" : "0")};
-  margin-right: ${(props) => (props.childLevel > 0 ? "1.5em" : "0")};
-  ${(props) => props.childLevel > 0 && "background-color: #fffcf0;"}
-  ${(props) =>
-    props.withdrawn && `text-decoration: line-through; color: ${props.theme.palette.grey[400]};`}
-`;
+interface ControlListOptions {
+  childLevel: number;
+}
+
+interface OSCALControlCardProps extends CardProps, ControlListOptions {
+  withdrawn: boolean;
+}
 
-function ControlsList(props) {
+const OSCALControlCard = styled(Card)<OSCALControlCardProps>(
+  ({ theme, childLevel, withdrawn }) => ({
+    marginTop: "1em",
+    marginBotom: "1em",
+    marginLeft: childLevel ? "1.5em" : "0",
+    marginRight: childLevel ? "1.5em" : "0",
+    backgroundColor: childLevel ? "#FFFCF0" : undefined,
+    textDecoration: withdrawn ? "line-through" : undefined,
+    color: withdrawn ? theme.palette.grey[400] : undefined,
+  })
+);
+
+interface ControlsListProps extends EditableFieldProps, AnchorLinkProps, ControlListOptions {
+  control: Control;
+  componentId?: string;
+  implementedRequirement?: ImplementedRequirementElement;
+  includeControlIds?: string[];
+  excludeControlIds?: string[];
+  onRestError?: (...args: any[]) => void;
+  onRestSuccess?: (...args: any[]) => void;
+  modificationAlters?: Alteration[];
+  modificationSetParameters?: SetParameterValue[];
+  withdrawn: boolean;
```

Comment:
`onRestSuccess` and `onRestError` probably need to be in another interface that this one extends.

Reply:
Going to do this in #830 instead...